### ISSUE TITLE
"date" inside an object in object column appears as and will be converted to string after clicking on dashboard

### ISSUE THREAD

**Issue by refre5h** [parse-community/parse-dashboard#285]
I believe this happened in the new dashboard in parse.com, for such reason I didn't use the new dashboard in parse.com. Now I found that it's also happening here with parse-server and parse-dashboard.
- [x] You're running version >=2.1.4 of Parse Server.
- [x] You've searched through [existing issues](https://github.com/ParsePlatform/parse-dashboard/issues?utf8=%E2%9C%93&q=). Chances are that your issue has been reported or resolved before.
#### Environment Setup

"express": "~4.13.4",
"kerberos": "~0.0.x",
"parse": "^1.8.0",
"parse-server": "~2.2.2",
"parse-dashboard": "~1.0.9"
Heroku
mlab
#### Steps to reproduce

Step 1.
Create a `object` column in any class

Step 2.
Write an object that has `date` in it to this column. Below is the exact copy of one found in our db (mlab).

``` javascript
   "billedSessions": {
        "0": {
            "startTime": {
                "__type": "Date",
                "iso": "2016-04-13T05:10:50.261Z"
            },
            "endTime": {
                "__type": "Date",
                "iso": "2016-04-13T05:11:30.118Z"
            },
            "minutes": 1
        }
    },
```

Step 3.
Use the client to get this value. The `startTime` and `endTime` are both `date` object (for now).

Step 4.
Go to parse-dashboard and find this value. Note that the date in the object appears as a string instead.

``` javascript
{"0":{"startTime":"2016-04-13T05:10:50.261Z","endTime":"2016-04-13T05:11:30.118Z","minutes":1}}
```

Step 5.
Double click this value to enter edit state, **but don't change anything**.

Step 6.
Focus somewhere else to exit the edit state.

Step 7.
Go back to mlab, check the value again, it has beome: 

``` javascript
    "billedSessions": {
        "0": {
            "startTime": "2016-04-13T05:10:50.261Z",
            "endTime": "2016-04-13T05:11:30.118Z",
            "minutes": 1
        }
    },
```

Step 8.
Use the client to get this value again, the type of `startTime` and `endTime` have both become `string`.

**Comment by drew-gross** [comment]
Yeah, our handling of non-raw-data for inner keys of objects isn't ideal. I'll see what we can do about this.

**Comment by Silkjaer** [comment]
Same goes for Date type in Config

**Comment by refre5h** [comment]
@Silkjaer A bit off topic, but is your dashboard Config working fine? It seems I can only add new params, which is saved successfully, but the Config page appears empty again if I refresh the dashboard.

**Comment by drew-gross** [comment]
It works for me. Can you try using the API console to see if you can see the config?

**Comment by refre5h** [comment]
@drew-gross Yes, using the API console can get the config.
```
GET
classes/_GlobalConfig
Yes on Use Master Key
```
```
{
    "results": [
        {
            "objectId": "1",
            "updatedAt": "2016-03-25T00:53:42.065Z",
            "params": {
                "announcementVersion": 1,
                "minimumSupportBuild": 1,
                ... // along with many other params
            },
            "createdAt": "2015-12-03T21:12:45.035Z"
        }
    ]
}
```
But the Config page appears empty.

**Comment by refre5h** [comment]
I fixed the Config issue by adding `javascriptKey` in the parse-dashboard-config.json. For anyone who's having the same issue, hope this helps. And sorry for going off topic again.

**Comment by kcoop** [comment]
Any progress with this? It's causing a crashing bug if I don't tread very carefully. I imagine it could bite a lot of people.

**Comment by TigPT** [comment]
I would like to add that its not possible by parse-dashboard UI to set a date back to (undefined). would be great if i could remove a date by the dashboard.

**Comment by douglasmuraoka** [comment]
Fixed by #1223
